src/components/games/seabattle/game.ts

```diff
@@ -54,23 +54,33 @@ export const SeabattleGame = Game({
       const player = parseInt(ctx.playerID, 10);
       // Avoid another salvo if the last was a miss.
       if (G.salvos.length > 0 &&
-          G.salvos[G.salvos.length - 1].player ===  player &&
-          !G.salvos[G.salvos.length - 1].hit) {
+        G.salvos[G.salvos.length - 1].player === player &&
+        !G.salvos[G.salvos.length - 1].hit) {
         return { ...G };
       }
-      const shipIndex = findShipWithCell(G.ships, {x, y}, player);
+      // Do not allow the same cells to be shot twice
+      const uniqueCell = G.salvos.filter((existG) => {
```

Comment:
It is a small detail, but makes the whole difference. Because you used curly braces { instead of (, you opened a new scope, which means that the interpreter will execute the code, not evaluate it. 
That means that it is totally ignoring the result of
        existG.player === player &&
        existG.cell.x === x &&
        existG.cell.y === y;
      

1- Replace {} braces with ()
2- existG is not a descriptive name, replace with salvo
3- add a unit test that tests this, because this bug almost went unnoticed.